src/log/log_inline.h

```diff
@@ -64,13 +64,13 @@ __wt_log_cmp(WT_LSN *lsn1, WT_LSN *lsn2)
 
 /*
  * __wt_lsn_string --
- *     Return a printable string representation of an lsn.
+ *     Return a printable string representation of an lsn into a fixed array.
  */
 static WT_INLINE int
-__wt_lsn_string(WT_SESSION_IMPL *session, WT_LSN *lsn, WT_ITEM *buf)
+__wt_lsn_string(WT_LSN *lsn, char *arr)
```

Comment:
I suggest having all the callers pass in the length of the buffer (I find `buf` easier to reason about than `arr`) via `sizeof(lsn_str)` or whatever it is named, and add an assert that `len` is >= `WT_MAX_LSN_STRING`.